web/modules/custom/rdf_entity_provenance/src/ProvenanceHelperInterface.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_entity_provenance;
+
+use Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Provides helper methods to fetch, check and update provenance data.
+ */
+interface ProvenanceHelperInterface {
+
+  /**
+   * Checks whether the the entity has provenance records.
+   *
+   * If an entity has provenance records it means that it is associated with an
+   * external source and thus has been federated.
+   *
+   * @param \Drupal\rdf_entity\RdfInterface $rdf_entity
+   *   The rdf entity.
+   *
+   * @return bool
+   *   Whether the entity has provenance record associated and thus, is
+   *   federated.
+   */
+  public function isFederated(RdfInterface $rdf_entity): bool;
```

Comment:
I'm a little bit reluctant about the added value of this method. As the implementation is saying, it's a simple "load by referred entity" which is accomplished by `::getProvenanceActivity()`